src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -246,8 +193,18 @@ var TestRenderer = ReactFiberReconciler({
   useSyncScheduling: true,
 
   getPublicInstance(ref) {
-    const createNodeMock = ref.rootContainerInstance.createNodeMock;
-    return createNodeMock(ref);
+    if (typeof ref.text === 'string') {
+      return ref.text;
+    } else if (
+      typeof ref.rootContainerInstance === 'object' &&
+      typeof ref.rootContainerInstance.createNodeMock === 'function'
+    ) {
+      const createNodeMock = ref.rootContainerInstance.createNodeMock;
+      return createNodeMock(ref);
+    }
+
+    // this should not be possible
+    throw new Error('Attempted to getPublicInstance on an invalid ref.');
```

Comment:
Not really important to use symbols there (if you don't leak those objects externally), you can just use an enum. Example [declaration](https://github.com/facebook/react/blob/1c43f9471773a5206279236c5e8b5b37d3e985cb/src/shared/ReactTypeOfWork.js#L15-L29) and [switch](https://github.com/facebook/react/blob/1c43f9471773a5206279236c5e8b5b37d3e985cb/src/renderers/shared/fiber/ReactFiberBeginWork.js#L596-L603).